File: android/src/main/java/top/huic/tencent_im_plugin/message/CustomMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.v2.V2TIMCustomElem;
import com.tencent.imsdk.v2.V2TIMManager;
import com.tencent.imsdk.v2.V2TIMMessage;

import top.huic.tencent_im_plugin.message.entity.CustomMessageEntity;

/**
 * 自定义消息节点
 */
public class CustomMessageNode extends AbstractMessageNode<V2TIMCustomElem, CustomMessageEntity> {

    @Override
    public V2TIMMessage getV2TIMMessage(CustomMessageEntity entity) {
        return V2TIMManager.getMessageManager().createCustomMessage(entity.getData().getBytes());
    }

    @Override
    public String getNote(V2TIMCustomElem elem) {
        return "[其它消息]";
    }

    @Override
    public CustomMessageEntity analysis(V2TIMCustomElem elem) {
        return new CustomMessageEntity(elem);
    }

    @Override
    public Class<CustomMessageEntity> getEntityClass() {
        return CustomMessageEntity.class;
    }
}

File: ios/Classes/message/CustomMessageNode.swift
import ImSDK

//
//  TextMessageNode.swift
//  tencent_im_plugin
//
//  Created by 蒋具宏 on 2020/3/13.
//  自定义消息节点
public class CustomMessageNode: AbstractMessageNode {

    override func getV2TIMMessage(params: [String: Any]) -> V2TIMMessage {
        let data: String = getParam(params: params, paramKey: "data")!;
        let desc: String? = getParam(params: params, paramKey: "desc");
        let ext: String? = getParam(params: params, paramKey: "ext");
        return V2TIMManager.sharedInstance().createCustomMessage(data.data(using: String.Encoding.utf8), desc: desc, extension: ext)
    }

    override func getNote(elem: V2TIMElem) -> String {
        "[其它消息]"
    }

    override func analysis(elem: V2TIMElem) -> AbstractMessageEntity {
        CustomMessageEntity(elem: elem as! V2TIMCustomElem)
    }
}
